### PR TITLE
Image loader support addons

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
This PR extends the image-load to optionally retag the images from the addons.
The docker code was extracted for better structuring.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: api/cmd/image-loader/main.go [hunk 1]
```diff
@@ -4,166 +4,159 @@ import (
...
+func processImages(ctx context.Context, log *zap.Logger, dryRun bool, images []string) error {
+	if err := docker.DownloadImages(ctx, log, dryRun, images); err != nil {
+		return fmt.Errorf("failed to download all images: %v", err)
 	}
 
-	if err = downloadImages(images); err != nil {
-		glog.Fatalf(err.Error())
-	}
-	retaggedImages, err := retagImages(registryName, images)
+	retaggedImages, err := docker.RetagImages(ctx, log, dryRun, images, "localhost:5000")
```
Comment (alvaroaleman) [inline api/cmd/image-loader/main.go]: Will this become configurable?

Also is this really worth adding a docker dependency?

File: api/cmd/image-loader/main.go [hunk 2]
```diff
@@ -391,16 +366,160 @@ func createNamedSecrets(secretNames []string) *corev1.SecretList {
...
+
+	// Statefulset
+	case *appsv1.StatefulSet:
+		return getImagesFromPodSpec(obj.Spec.Template.Spec)
+	case *appsv1beta1.StatefulSet:
+		return getImagesFromPodSpec(obj.Spec.Template.Spec)
+	case *appsv1beta2.StatefulSet:
+		return getImagesFromPodSpec(obj.Spec.Template.Spec)
+
+		// DaemonSet
```
Comment (alvaroaleman) [inline api/cmd/image-loader/main.go]: Comment indentation is wrong (here and below)

File: api/cmd/image-loader/main.go (hunk 1 above)
Comment (mrIncompetent) [reply api/cmd/image-loader/main.go]: > Will this become configurable?

Missed removing that :man_facepalming: 

> Also is this really worth adding a docker dependency?

Actually its our own package there - Just moved the docker interaction into a package for better structuring.
The imported docker dependency is just for the purpose of parsing image names (They have a really good library for that purpose)

File: api/cmd/image-loader/main.go (hunk 2 above)
Comment (mrIncompetent) [reply api/cmd/image-loader/main.go]: Fixed with https://github.com/kubermatic/kubermatic/pull/3666/commits/519fd336b751a1f40d83aa11b3cfd7b5ce4d0d25

**Comment by mrIncompetent** [comment]
@alvaroaleman PTAL

**Pull request merged**